### DISCUSSION TITLE
Why does `SPC v` not translate to `C-v`?

### DISCUSSION THREAD

**Discussion by joelnoomi** [meow-edit/meow#338] [Q&A] [answered]
In TUTORIAL.org on line [80](https://github.com/meow-edit/meow/blame/master/TUTORIAL.org#L80) and [92](https://github.com/meow-edit/meow/blame/master/TUTORIAL.org#L92), it says, respectively:

> - Entering KEYPAD state by pressing `SPC` in NORMAL or MOTION state.
...
> - Any other `<key>` will be translated to `C-<key>`.

Based on that, I would have expected `SPC v` to be translated to `C-v` and thus run the command `scroll-up-command`. But instead of scrolling up, the character `v` is inserted at point.

When inspecting the output from `view-lossage`, I observe the following:

```
 SPC		;; meow-keypad
 v v		;; meow-keypad-self-insert
```

If I, for example, do `SPC h`, on the other hand, that brings me to `KEYPAD: C-h`. Why does the same not happen for `SPC v`?

Is this a misunderstanding on my part, inaccurate documentation, or possibly even a bug? Most likely it's the first and, in that case, an explanation would be very helpful and much appreciated.

**Comment by mmarshall540** [comment]
I think the tutorial needs to be updated.  This was previously how it worked, but it has changed..

Now, SPC is known as the "leader" key.  It brings up a merged keymap, consisting of the `mode-specific-map` (user bindings normally reached by pressing "C-c"), plus additional bindings for "x", "c", "h", "m", and "g".

You can still use "m" to prefix the next key with "Meta" modifier, and you can still use "g" to prefix the next key with "Ctrl+Meta".  Also, "x" enters the "C-x" prefix, and so on for "c" and "h".

Other keys are handled literally, so for example "SPC (" inserts a parenthesis.

After pressing "x", "c", or "h" any other keys will be handled as if pressed with "Ctrl", unless there is no such binding, in which case, it will be without "Ctrl".  So "SPC x s" is equivalent to "C-x C-s".

If you want "SPC v" to be equivalent to "C-v" again, you can do it like this:

    (meow-leader-define-key ("v" . "C-v"))

In similar fashion, you can pick which commands you use most frequently and add those as well.  It's also useful for commands bound to "C-x \<key\>", since those would otherwise require pressing "SPC x SPC \<key\>".

**Comment by eshrh** [comment]
@mmarshall540 is right.

As things stand at the moment, when you define a key for the leader, it gets registered into `C-c`. So, running the code they mentioned will also lead to `C-c v` being bound to `C-v`. The idea behind this is that leader keybinds should never be inaccessible, even when meow is off.

I'm not sure when SPC-v was C-v . At least, at the time of the commit that added those lines to the tutorial, 845c4693ade5aa24715d3c0e4da678ca3fa98b51, SPC v still self inserted v. I assume the docs need to fixed as well.

**Reply by mmarshall540** [reply]
I could be mistaken that "SPC v" used to do "C-v".  It was my impression, but I'm not sure if that came from actual experience or rereading the documentation.

**Reply by DogLooksGood** [reply]
`SPC v` should be translated to `C-c v`. 

In fact, `SPC` is the key used to turn on KEYPAD state. In keypad state, keys will be translated. So `SPC c c` = Turn on Keypad, `C-c`, `C-c`.
And since it's pointless to have every keys get translated(e.g. `C-a`, `C-e`, `C-n`, `C-p`, etc), we introduced a fallback layer, that is the leader key layer.

This leader layer will translate `<letter>` to `C-c <letter>`. And `C-c <letter>` is the user custom space in vanilla Emacs.

**Comment by eshrh** [comment, marked as answer]
I finally understood what the docs were supposed to mean. In the old source, that bullet was dependent on the bullet point above it. Just rewrote the section in d180d47c56ec06d054fb101bc843340d93c26d41
